src/lib.rs

```diff
@@ -151,7 +151,12 @@ pub struct PurchaseResponse {
 pub trait Validator: Send + Sync {
     /// Called to perform the validation on whichever platform is described in the provided UnityPurchaseReceipt.
     async fn validate(&self, receipt: &UnityPurchaseReceipt) -> Result<PurchaseResponse>;
-    /// Similar to the helper function `crate::get_apple_receipt_data`, an associated function for pulling the response from owned secrets.
+}
+
+/// Trait which allows us to retrieve receipt data from an object's own secrets.
+#[async_trait]
+pub trait ReceiptDataGetter {
```

Comment:
Do you know David *Getter*? JK
Maybe Margaret Fetcher? :)

Reply:
LOL

Reply:
Should I change the name of the methods to match? It's going to be a breaking update anyway

Reply:
yeah lets do that